test/patterns.jl

```diff
@@ -119,4 +119,40 @@
             @test Pattern(:_, :__, :input, :__) != reduced
         end
     end
+
```

Comment:
do you want to test the `show` method here?

Reply:
It'd be a pretty basic test and the doctests are already covering it. I was actually considering dropping testing the `show` method for `KeyedDataset`.